### ISSUE TITLE
a way to denote that a decorator uses it's function

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#538]
```py
def register[Fn: Callable[[], None]](fn: Fn) -> Fn:
    register._fn = fn

@register
def _foo():  # unused
    print("hi")

register._fn()
```

when i have a 'private' function that is decorated with with some decorator that performs some action, either in place or saves it for later, it will get crowdstriked (n.b. we now call false positives "crowdstrikes") with `reportUnusedFunction`

can an `Annotated` be introduced:
```py
from basedtyping import Used

def register[Fn: Callable[[], None]](fn: Annotated[Fn, Used]) -> Fn:
    register._fn = fn
```

**Comment by KotlinIsland** [comment]
while we are there we can add a `Calls`, which means that the function will be called in place
